pkg/sriov/network.go

```diff
@@ -90,6 +90,25 @@ func (builder *NetworkBuilder) WithVLAN(vlanID uint16) *NetworkBuilder {
 	return builder
 }
 
+// WithQinQSupport sets the VLAN protocol for qinq tunneling protocol in the SrIovNetwork definition spec.
+func (builder *NetworkBuilder) WithQinQSupport(vlanProtocol string) *NetworkBuilder {
+	if valid, _ := builder.validate(); !valid {
+		return builder
+	}
+
+	glog.V(100).Infof("Setting SriovNetwork vlanProtocol support for qinq")
+
+	if vlanProtocol == "" {
```

Comment:
please change validation to something specific:
```
	allowedVlanProto := []string{"802.1q","802.1Q", "802.1ad", "802.1AD"}
	if !slices.Contains(allowedVlanProto, vlanProtocol) {
		builder.errorMsg = "invalid 'vlanProtocol' parameters"
	}
```

Reply:
changed